nx/lib/nx/binary_backend.ex

```diff
@@ -1082,6 +1083,186 @@ defmodule Nx.BinaryBackend do
     {q, r}
   end
 
+  @impl true
+  def svd(
+        {u_holder, %{shape: {m, n} = s_shape, type: output_type} = s_holder, v_holder},
+        %{type: input_type} = tensor
+      ) do
+    # This implementation is a mixture of concepts described in [1] and the
+    # algorithmic descriptions found in [2] and [3]
+    # [1] -
+    #    Parallel One-Sided Block Jacobi SVD Algorithm: I. Analysis and Design,
+    #    by Gabriel Oksa and Marian Vajtersic
+    #    Source: https://www.cosy.sbg.ac.at/research/tr/2007-02_Oksa_Vajtersic.pdf
+    # [2] - https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/client/lib/svd.cc#L784
+    # [3] - https://github.com/tensorflow/tensorflow/blob/dcdc6b2f9015829cde2c02b111c04b2852687efc/tensorflow/compiler/xla/client/lib/svd.cc#L386
+
+    a_bin = to_binary(tensor)
+
+    {u_bin, d_bin, v_bin} = householder_bidiagonalization(a_bin, s_shape, input_type, output_type)
+
+    max_iter = 100
+
+    {fro_norm, _diag_norm, off_diag_norm} = get_matrix_norms(d_bin, s_shape, output_type)
+
+    Enum.reduce_while(1..max_iter, {u_bin, d_bin, v_bin, off_diag_norm, fro_norm}, fn _,
+                                                                                      {u_bin,
+                                                                                       d_bin,
+                                                                                       v_bin,
+                                                                                       off_diag_norm,
+                                                                                       fro_norm} ->
```

Comment:
You have two options here so the formatter doesn't give you a hard time.

1. Move the accumulator to its own line:

```suggestion
    acc = {u_bin, d_bin, v_bin, off_diag_norm, fro_norm}
    Enum.reduce_while(1..max_iter, acc, fn _, {u_bin, d_bin, v_bin, off_diag_norm, fro_norm} ->
```

2. Break the function head into its own clause:

```suggestion
    Enum.reduce_while(1..max_iter, {u_bin, d_bin, v_bin, off_diag_norm, fro_norm}, fn
      _, {u_bin, d_bin, v_bin, off_diag_norm, fro_norm} ->
```

WDYT?

Reply:
Looks a lot better!